### PR TITLE
Add some more tests for DelayQueue and DelayFIFO

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

**Comment by sttts** [comment]
@jdef can you take a look?

File: pkg/queue/delay_test.go [hunk 1]
```diff
@@ -112,6 +129,37 @@ func TestDQ_sanity_check(t *testing.T) {
...
+func TestDQ_Offer(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+
+	dq := NewDelayQueue()
+	delay := time.Second
+
+	added := dq.Offer(&testjob{})
+	if added {
+		t.Fatalf("offered job without deadline added")
```
Comment (joerg84) [inline pkg/queue/delay_test.go]: Tiny nit: Could we change the log message to something like "DelayQueue should not accept job without deadline"

**Comment by joerg84** [comment]
One tiny nit, otherwise lgtm.

**Comment by jdef** [comment]
LGTM

**Comment by sttts** [comment]
@joerg84 improved the error messages.

In fact, I much prefer the assert or expect style of assumption. Then one usually gets proper messages for free. Using the assert module for all new tests now.

**Comment by joerg84** [comment]
@sttts Thanks, error messages look much better!

**Pull request merged**